lumicks/pylake/tests/test_imaging_confocal/test_scan.py

```diff
@@ -478,3 +479,175 @@ def test_plot_single_channel_percentile_color_adjustment(test_scans):
+        [slice(None), slice(0, None), slice(1, None)],
+        [slice(None), slice(1, 4), slice(1, 5)],
+        [slice(None), slice(1, 3), slice(None)],
+        [slice(None), slice(None), slice(1, 3)],
+        [slice(None), slice(1, 2), slice(1, 2)],  # Single pixel
+        [slice(None), slice(1, 2), slice(None)],
+        [slice(None), slice(None), slice(1, 2)],
+        [0],  # Only indexing scans
+        [0, slice(1, 2), slice(1, 2)],
+        [-1, slice(1, 3), slice(None)],
+        [0, slice(None), slice(1, 2)],
+        [0, slice(1, 3)],  # This tests a very specific corner case where _num_pixels could fail
+    ],
+)
+def test_scan_get_item_slicing(all_slices, test_scans):
+    """Test slicing, slicing is given as image, y, x"""
+
+    valid_scans = [
+        "fast Y slow X multiframe",
+        "fast Y slow X",
+        "fast X slow Z multiframe",
+        "fast Y slow Z multiframe",
+        "red channel missing",
+        "rb channels missing",
+    ]
+
+    for key in valid_scans:
+        scan = test_scans[key]
+
+        # Slice how we would with numpy
+        slices = tuple(all_slices if scan.num_frames > 1 else all_slices[1:])
+        cropped_scan = scan[all_slices]
+        np.testing.assert_allclose(cropped_scan.timestamps, scan.timestamps[slices])
+
+        for channel in ("rgb", "green"):
+            ref_img = scan.get_image(channel)[slices]
+            np.testing.assert_allclose(cropped_scan.get_image(channel), ref_img)
+        # Numpy array is given as Y, X, while number of pixels is given sorted by spatial axis
+        # i.e. X, Y
+        np.testing.assert_allclose(
+            np.flip(cropped_scan._num_pixels), cropped_scan.get_image("green").shape[-2:]
+        )
+
+
+def test_slicing_cropping_separate_actions(test_scans):
+    """Test whether cropping works in a sequential fashion"""
+    multi_frame, single_frame = (
+        generate_scan(
+            "test",
+            np.random.rand(num_frames, 5, 7),
+            [1, 1],
+            start=0,
+            dt=100,
+            samples_per_pixel=1,
+            line_padding=1,
+        )
+        for num_frames in (8, 1)
+    )
+
+    def assert_equal(first, second):
+        np.testing.assert_allclose(first.get_image("red"), second.get_image("red"))
+        np.testing.assert_allclose(first.get_image("rgb"), second.get_image("rgb"))
+        assert first.num_frames == second.num_frames
+        assert first._num_pixels == second._num_pixels
+
+    assert_equal(multi_frame[1:3][:, 1:3, 2:3], multi_frame[1:3, 1:3, 2:3])
+    assert_equal(multi_frame[1][:, 1:3, 2:3], multi_frame[1, 1:3, 2:3])
+    assert_equal(multi_frame[:, 1:3, 2:3][3], multi_frame[3, 1:3, 2:3])
+    assert_equal(multi_frame[:, 1:, 2:][3], multi_frame[3, 1:, 2:])
+    assert_equal(multi_frame[:, :3, :3][3], multi_frame[3, :3, :3])
+    assert_equal(single_frame[0][:, 1:5, 2:6][:, 1:3, 2:8], single_frame[0, 2:4, 4:6])
+
+
+def test_error_handling_multidim_indexing():
```

Comment:
missing a test for slicing with `list`, which should raise `IndexError("Only slicing by int or slice is allowed to select frames")`, but connected with my comment about this above

Reply:
Indeed. Added a slightly more general test which just flags that whatever type you sliced with isn't allowed.